sklearn/datasets/openml.py

```diff
@@ -263,6 +268,92 @@ def _convert_arff_data(arff_data, col_slice_x, col_slice_y, shape=None):
         raise ValueError('Unexpected Data Type obtained from arff.')
 
 
+def _feature_to_dtype(feature):
+    """Map feature to dtype for pandas DataFrame
+    """
+    if feature['data_type'] == 'string':
+        return object
+    elif feature['data_type'] == 'nominal':
+        return 'category'
+    # only numeric, integer, real are left
+    elif (feature['number_of_missing_values'] != '0' or
+          feature['data_type'] in ['numeric', 'real']):
+        # cast to floats when there are any missing values
+        return np.float64
+    elif feature['data_type'] == 'integer':
+        return np.int64
+    raise ValueError('Unsupported feature: {}'.format(feature))
+
+
+def _chunk_generator(gen, chunksize):
```

Comment:
I think we have a very similar util already, or do we only have for arrays?

Reply:
Maybe we can use `get_chunk_n_rows` and `gen_batches` for this.

Reply:
`gen_batches` needs the number of rows. In this case, the number of rows may not be `None`.

Reply:
I would move this function in the utils next to the other generator.
I don't think that we can factorize it with the current one.